lib/Tool/MaintenanceModeHelper.php

```diff
@@ -0,0 +1,92 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Tool;
+
+use Doctrine\DBAL\Exception\TableNotFoundException;
+use Pimcore\Event\SystemEvents;
+use Pimcore\Model\Tool\TmpStore;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class MaintenanceModeHelper implements MaintenanceModeHelperInterface
+{
+    protected const ENTRY_ID = "maintenace_mode";
+
+    public function __construct(protected RequestStack $requestStack)
+    {
+    }
+
+    public function activate(string $sessionId): void
+    {
+        if (empty($sessionId)) {
+            $sessionId = $this->requestStack->getSession()->getId();
+        }
+
+        if (empty($sessionId)) {
+            throw new \InvalidArgumentException("Pass sessionId to activate the maintenance mode");
+        }
+
+        $this->addEntry($sessionId);
+
+        \Pimcore::getEventDispatcher()->dispatch(new GenericEvent(), SystemEvents::MAINTENANCE_MODE_ACTIVATE);
+    }
+
+    public function deactivate(): void
+    {
+        $this->removeEntry();
+
+        \Pimcore::getEventDispatcher()->dispatch(new GenericEvent(), SystemEvents::MAINTENANCE_MODE_DEACTIVATE);
+    }
+
+    public function isActive(string $matchSessionId = null): bool
+    {
+        if ($maintenanceModeEntry = $this->getEntry()) {
+            if ($matchSessionId && $matchSessionId !== $maintenanceModeEntry) {
+                return false;
+            }
+            return true;
```

Comment:
the logic here was inverted, if the session id matches, for me maintenance mode should be not active (false)

```suggestion
            if ($matchSessionId && $matchSessionId === $maintenanceModeEntry) {
                return false;
            }
            return true;
```
or [invert the return ](https://github.com/pimcore/pimcore/pull/15954/files/49112a8d37431842410ded9fd4954ec6218ee38d?w=1#r1330470184)

Reply:
Fixed with merging the or suggestion